pyonfx/timestamps.py

```diff
@@ -0,0 +1,552 @@
+            if lowest_timestamp is None or lowest_timestamp > timestamp:
+                lowest_timestamp = timestamp
+
+            rounded_timestamp = rounding_method(timestamp)
+
+            if version == 2 and rounded_timestamp < previous_timestamp:
+                raise ValueError(
+                    f"The timestamps file contain timestamps NOT in ascending order."
+                )
+
+            previous_timestamp = rounded_timestamp
+            timestamps.append(rounded_timestamp)
+
+        if not len(timestamps):
+            raise ValueError(f"The timestamps file is empty.")
+
+        if version == 4:
+            timestamps.sort()
+
+        duration = highest_timestamp - lowest_timestamp
+        if duration:
+            fpms = Fraction(len(timestamps) - 1) / (
+                highest_timestamp - lowest_timestamp
+            )
+        else:
+            fpms = 0
+        return timestamps, highest_timestamp, fpms
+
+
+class Timestamps:
+    """Timestamps object contains informations about the timestamps of an video.
+
+    Video player have 2 methods to deal with timestamps. Some floor them and other round them.
+    This can lead to difference when displaying the subtitle.
+        Ex:
+            Player - Method - proof
+            mpv    - round  - https://github.com/mpv-player/mpv/blob/7480efa62c0a2a1779b4fdaa804a6512aa488400/sub/sd_ass.c#L499
+            FFmpeg - floor  - https://github.com/FFmpeg/FFmpeg/blob/fd1712b6fb8b7acc04ccaa7c02b9a5c9f233cfb3/libavfilter/vf_subtitles.c#L195
+            VLC    - floor  - https://github.com/videolan/vlc/blob/f7bb59d9f51cc10b25ff86d34a3eff744e60c46e/include/vlc_tick.h#L118-L130
+            MPC-HC - floor  - https://github.com/clsid2/mpc-hc/blob/0994fd605a9fb4d15806d0efdd6399ba1bc5f984/src/Subtitles/LibassContext.cpp#L843
+    Important note:
+        Matroska (.mkv) file are an exception !!!
+        If you want to be compatible with mkv, use RoundingMethod.ROUND.
+        By default, they only have a precision to milliseconds instead of nanoseconds like most format.
+            For more detail see:
+                1- https://mkvtoolnix.download/doc/mkvmerge.html#mkvmerge.description.timestamp_scale
+                2- https://matroska.org/technical/notes.html#timestampscale-rounding
+
+    Parameters:
+        rounding_method (RoundingMethod): A rounding method. See the comment above about floor vs round.
+            99% of the time, you want to use RoundingMethod.ROUND.
+        timestamps (List[int], optional): A list of [timestamps](https://en.wikipedia.org/wiki/Timestamp) in milliseconds encoded as integers.
+                                It represent each frame [presentation timestamp (PTS)](https://en.wikipedia.org/wiki/Presentation_timestamp)
+        normalize (bool, optional): If True, it will shift the timestamps to make them start from 0. If false, the option does nothing.
+        fpms (Fraction, optional): The fpms.
```

Comment:
I did that because ``from_timestamps_file`` and ``from_video_file`` deal with milliseconds.
But, we can change ``fpms`` to ``fps`` in the constructor, but it will create unnecessary calculations
Ex:
``from_timestamps_file`` will create a ``fpms``. It will convert it to ``fps`` then when it call the constructor, it will reconvert it to ``fpms``

Reply:
Ahhhh, I think it is more because the methods "ms_to_frames" and "frames_to_ms" works with ms. So we can have everything in ms this way. We can keep it as you did, I would just specify what "fpms" is (frames per milliseconds) directly in the doc.

Reply:
Done